docs/ops/deployment/local.md

```diff
@@ -0,0 +1,178 @@
+---
+title: "Local Cluster"
+nav-title: 'Local Cluster'
+nav-parent_id: deployment
+nav-pos: 1
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+Get a local Flink cluster up and running in a few simple steps.
+
+* This will be replaced by the TOC
+{:toc}
+
+## Setup: Download and Start Flink
+
+Flink runs on __Linux, Mac OS X, and Windows__.
+To be able to run Flink, the only requirement is to have a working __Java 8.x__ installation.
+
+You can check the correct installation of Java by issuing the following command:
+
+{% highlight bash %}
+java -version
+{% endhighlight %}
+
+If you have Java 8, the output will look something like this:
+
+{% highlight bash %}
+java version "1.8.0_111"
+Java(TM) SE Runtime Environment (build 1.8.0_111-b14)
+Java HotSpot(TM) 64-Bit Server VM (build 25.111-b14, mixed mode)
+{% endhighlight %}
+
+{% if site.is_stable %}
+<div class="codetabs" markdown="1">
+<div data-lang="Download and Unpack" markdown="1">
+1. Download a binary from the [downloads page](https://flink.apache.org/downloads.html). You can pick
+   any Scala variant you like. For certain features you may also have to download one of the pre-bundled Hadoop jars
+   and place them into the `/lib` directory.
+2. Go to the download directory.
+3. Unpack the downloaded archive.
+
+{% highlight bash %}
+$ cd ~/Downloads        # Go to download directory
+$ tar xzf flink-*.tgz   # Unpack the downloaded archive
+$ cd flink-{{site.version}}
+{% endhighlight %}
+</div>
+
+<div data-lang="MacOS X" markdown="1">
+For MacOS X users, Flink can be installed through [Homebrew](https://brew.sh/).
+
+{% highlight bash %}
+$ brew install apache-flink
+...
+$ flink --version
+Version: 1.2.0, Commit ID: 1c659cf
+{% endhighlight %}
+</div>
+
+</div>
+
+{% else %}
+### Download and Compile
+Clone the source code from one of our [repositories](https://flink.apache.org/community.html#source-code), e.g.:
+
+{% highlight bash %}
+$ git clone https://github.com/apache/flink.git
+$ cd flink
+$ mvn clean package -DskipTests # this will take up to 10 minutes
```

Comment:
I think this takes more than 10 minutes by now ;-)

One way to further accelerate is to use `mvn clean package -DskipTests -Pfast`
@zentol

Reply:
yeah it's more like 25 I think. Depending on the environment the build of the WebUI may also slow things down significantly; or fail outright. May be useful to document `-Pskip-webui-build` for these cases.

Reply:
Fair point. WSL cannot build the web-ui at the moment, simply deadlocks at some point :-(